packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -3,21 +3,105 @@
  * Licensed under the MIT License.
  */
 
+import assert from "node:assert";
+
 import {
 	type IMemoryTestObject,
 	benchmarkMemory,
 	isInPerformanceTestingMode,
 } from "@fluid-tools/benchmark";
 
 import { SharedMatrix } from "../../index.js";
+import { UndoRedoStackManager } from "../undoRedoStackManager.js";
 import { createLocalMatrix } from "../utils.js";
 
+/**
+ * Creates a benchmark for undo operations on a SharedMatrix.
+ */
+function createUndoBenchmark({
+	title,
+	matrixSize,
+	initialValue,
+	operation,
+	count,
+}: {
+	title: string;
+	matrixSize: number;
+	initialValue: string;
+	operation: (matrix: SharedMatrix, count: number) => void;
+	count: number;
+}): IMemoryTestObject {
+	return new (class implements IMemoryTestObject {
+		readonly title = title;
+		private localMatrix: SharedMatrix | undefined;
+		private undoStack: UndoRedoStackManager | undefined;
+
+		async run(): Promise<void> {
+			assert(this.undoStack !== undefined, "undoStack is not initialized");
+			for (let i = 0; i < count; i++) {
+				this.undoStack.undoOperation();
+			}
+		}
+
+		beforeIteration(): void {
+			this.localMatrix = createLocalMatrix({
+				id: "testLocalMatrix",
+				size: matrixSize,
+				initialValue,
+			});
+			this.undoStack = new UndoRedoStackManager();
+			this.localMatrix.openUndo(this.undoStack);
+			operation(this.localMatrix, count);
+		}
+	})();
+}
+
+/**
+ * Creates a benchmark for redo operations on a SharedMatrix.
+ */
+function createRedoBenchmark({
+	title,
+	matrixSize,
+	initialValue,
+	operation,
+	count,
+}: {
+	title: string;
+	matrixSize: number;
+	initialValue: string;
+	operation: (matrix: SharedMatrix, count: number) => void;
+	count: number;
+}): IMemoryTestObject {
+	return new (class implements IMemoryTestObject {
+		readonly title = title;
+		private localMatrix: SharedMatrix | undefined;
+		private redoStack: UndoRedoStackManager | undefined;
+
+		async run(): Promise<void> {
+			assert(this.redoStack !== undefined, "redoStack is not initialized");
```

Comment:
Is there a way for us to assert on the number of operations on the redo stack? Theoretically it should be `count`, and if it isn't, then something is wrong, right?

Reply:
We can add it, if needed. We don't export `UndoRedoStackManager`, so adding new properties is fine.

Reply:
Added new assertions on undostack/redostack